### PR TITLE
Extract test metadata for watch requests

### PR DESCRIPTION
**Pull request by dgoodwin**
[TRT-721](https://issues.redhat.com//browse/TRT-721)

### CODE REVIEW

File: pkg/prowloader/testoutputmetadata.go [hunk 1]
```diff
@@ -123,13 +129,17 @@ func scanLine(regex *regexp.Regexp, tokenKeys []string, line string) []map[strin
 			lt = strings.ReplaceAll(lt, "}", " ")
 			tokens := strings.Split(lt, " ")
 			for _, t := range tokens {
-				t = strings.TrimSuffix(t, ",")
+				t = strings.TrimSuffix(t, ",") // i.e. trim the trailing comma after: reason=foobar,
+				//t = strings.TrimSuffix(t, "\"") // i.e. trim the trailing comma after: reason=foobar"
```
Comment (DennisPeriquet) [inline pkg/prowloader/testoutputmetadata.go:133]: unnecessary comment?

File: pkg/prowloader/testoutputmetadata.go [hunk 2]
```diff
@@ -123,13 +129,17 @@ func scanLine(regex *regexp.Regexp, tokenKeys []string, line string) []map[strin
...
 				st := strings.Split(t, "=")
 				if len(st) == 2 && tokenKeyMap[st[0]] {
 					// Remove quotes around value if present:
 					v := st[1]
-					if strings.HasPrefix(v, "\"") && strings.HasSuffix(v, "\"") {
-						v = v[1 : len(v)-1]
+					if strings.HasPrefix(v, "\"") {
+						v = v[1:len(v)]
```
Comment (DennisPeriquet) [inline pkg/prowloader/testoutputmetadata.go:139]: this works but you can also use `v = v[1:]`

Comment (dgoodwin) [reply pkg/prowloader/testoutputmetadata.go:139]: Nice thank you, I knew of that in Python but didn't think Go would have it.

**Comment by dgoodwin** [comment]
Thanks @DennisPeriquet , feedback addressed

**Pull request merged**